app/src/main/kotlin/compiler/semantic_analysis/VariablePropertiesAnalyzer.kt

```diff
@@ -26,74 +26,63 @@ object VariablePropertiesAnalyzer {
+                        )
+                    }
+                    analyzeVariables(node.value, currentFunction)
                 }
-                is Statement.Block -> node.block.forEach { analyzeVariables(it, currentOwner) }
+                is Statement.Block -> node.block.forEach { analyzeVariables(it, currentFunction) }
                 is Statement.Conditional -> (
                     sequenceOf(node.condition) +
                         node.actionWhenTrue.asSequence() + (node.actionWhenFalse?.asSequence() ?: emptySequence())
-                    ).forEach { analyzeVariables(it, currentOwner) }
+                    ).forEach { analyzeVariables(it, currentFunction) }
                 is Statement.Loop -> (sequenceOf(node.condition) + node.action.asSequence())
-                    .forEach { analyzeVariables(it, currentOwner) }
-                is Statement.FunctionReturn -> analyzeVariables(node.value, currentOwner)
+                    .forEach { analyzeVariables(it, currentFunction) }
+                is Statement.FunctionReturn -> analyzeVariables(node.value, currentFunction)
                 is Expression.Variable -> {
-                    resolve(node, currentOwner, false)
+                    val resolvedVariable: NamedNode = nameResolution[node]!!
+                    variableProperties[resolvedVariable]!!.accessedIn.add(currentFunction!!)
                 }
-                is Expression.FunctionCall -> node.arguments.forEach { analyzeVariables(it, currentOwner) }
-                is Expression.FunctionCall.Argument -> analyzeVariables(node.value, currentOwner)
-                is Expression.UnaryOperation -> analyzeVariables(node.operand, currentOwner)
+                is Expression.FunctionCall -> node.arguments.forEach { analyzeVariables(it, currentFunction) }
+                is Expression.FunctionCall.Argument -> analyzeVariables(node.value, currentFunction)
+                is Expression.UnaryOperation -> analyzeVariables(node.operand, currentFunction)
                 is Expression.BinaryOperation -> sequenceOf(node.leftOperand, node.rightOperand)
-                    .forEach { analyzeVariables(it, currentOwner) }
+                    .forEach { analyzeVariables(it, currentFunction) }
                 is Expression.Conditional -> sequenceOf(node.condition, node.resultWhenTrue, node.resultWhenFalse)
-                    .forEach { analyzeVariables(it, currentOwner) }
+                    .forEach { analyzeVariables(it, currentFunction) }
                 is Function -> {
                     node.parameters.forEach {
-                        variableProperties.put(it, VariableProperties(node, false))
+                        variableProperties[it] = VariableProperties(node)
                         // scope of the inner function has not begun yet
-                        it.defaultValue?.let { analyzeVariables(it, currentOwner) }
+                        it.defaultValue?.let { analyzeVariables(it, currentFunction) }
```

Comment:
![image](https://user-images.githubusercontent.com/59764862/202906483-779d64d2-03bd-49a2-9f93-641259c86d16.png)
Using the same name `it` in the outer and the inner lambda may be a bit misleading